src/Root16.Sprout.Dataverse/DataSources/Dataverse/DataverseDataSource.cs

```diff
@@ -123,58 +132,56 @@ private static void RemoveAttribute(IEnumerable<DataOperation<Entity>> operation
             }
             else
             {
-                List<OrganizationRequestCollection> ListofRequestCollections = [];
-
-                foreach (var request in requestCollection.Chunk(1000))
-                {
-                    var orgRequestCollection = new OrganizationRequestCollection();
-                    orgRequestCollection.AddRange(request);
-                    ListofRequestCollections.Add(orgRequestCollection);
-                }
+                ConcurrentBag<DataOperationResult<Entity>> parallelResults = [];
 
-                List<Task<OrganizationResponse>> requestTasks = [];
+                ParallelOptions parallelOptions = new() { MaxDegreeOfParallelism = CrmServiceClient.RecommendedDegreesOfParallelism };
 
-                foreach (var requests in ListofRequestCollections)
+                await Parallel.ForEachAsync(requestCollection.Chunk(10), parallelOptions, async (batch, token) =>
                 {
-                    requestTasks.Add(CrmServiceClient.ExecuteAsync(new ExecuteMultipleRequest
+                    if (batch.Count() == 1)
+                    {
+                        var response = await TryExecuteRequestAsync(batch.Single());
+                        results.Add(ResultFromRequestType(batch.Single(), true));
+                    }
```

Comment:
@jrpetersjr I realized this single request doesn't have the same built-in error handling that EMR does, so this could potentially throw an unhandled exception for validation errors. I'm going to remove this check and prefer that single requests here still be sent within an EMR to simplify error handling.